pkg/apply/applier.go

```diff
@@ -140,7 +140,13 @@ func (a *Applier) Run(ctx context.Context) <-chan Event {
 		// This provides us with a slice of all the objects that will be
 		// applied to the cluster.
 		infos, _ := a.ApplyOptions.GetObjects()
-		err := a.ApplyOptions.Run()
+		var err error
+		// apply only if Destroy flag is off. Else skip the apply step and proceed to prune.
+		if !a.PruneOptions.Destroy {
+			err = a.ApplyOptions.Run()
+		} else {
+			err = nil
```

Comment:
This is not needed. `err` is already set to nil by default on line 143

Reply:
I wanted it to be explicit and call it out so that it's more readable.